### PR TITLE
rust-analyzer: use in-tree `pattern_analysis` crate

### PR DESCRIPTION
**Pull request by Nadrieril**
r? @Veykril

### CODE REVIEW

File: src/tools/rust-analyzer/crates/hir-ty/src/lib.rs [hunk 1]
```diff
@@ -15,9 +15,11 @@ extern crate rustc_abi;
...
 extern crate ra_ap_rustc_abi as rustc_abi;
 
-// Use the crates.io version unconditionally until the API settles enough that we can switch to
-// using the in-tree one.
-extern crate ra_ap_rustc_pattern_analysis as rustc_pattern_analysis;
+#[cfg(feature = "in-rust-tree")]
+extern crate rustc_pattern_analysis;
+
+#[cfg(not(feature = "in-rust-tree"))]
+extern crate ra_ap_rustc_pattern_analysis as rrustc_pattern_analysisustc_pattern_analysis;
```
Comment (fbstj) [inline src/tools/rust-analyzer/crates/hir-ty/src/lib.rs:22]:
```suggestion
extern crate ra_ap_rustc_pattern_analysis as rustc_pattern_analysis;
```
looks like a typo?

Comment (Nadrieril) [reply src/tools/rust-analyzer/crates/hir-ty/src/lib.rs:22]: oops x) ty

**Comment by Veykril** [comment]
> Failed to set assignee to `Veykril`: invalid assignee
> 
> > **Note**: Only org members with at least the repository "read" role, users with write permissions, or people who have commented on the PR may be assigned.

Shouldn't I qualify for that first one? 🤔

**Comment by matthiaskrgr** [comment]
@bors r- 
needs rebase/ does not build

**Comment by Veykril** [comment]
https://github.com/rust-lang/rust-analyzer/pull/17853 fixes the mismatches on our side now, so we should probably switch this to in-tree once that is merged and we've synced the subtree again

**Comment by Nadrieril** [comment]
Makes sense. Marking as draft in the meantime.

**Comment by lnicola** [comment]
The sync PR got merged today, we should be good to go?

**Comment by Nadrieril** [comment]
Oh, perfect!

**Review by lnicola — APPROVED** [review]

**Pull request merged**